modules/internal/DefaultRectangular.chpl

```diff
@@ -1428,15 +1453,36 @@ module DefaultRectangular {
             writeln("reallocating in-place");
 
           sizesPerDim(0) = reallocD.dsiDim(0).sizeAs(int);
-          data = _ddata_reallocate(data, eltType, oldSize, newSize);
+
+          if _arrayResizePolicy == chpl_ArrayResizePolicy.Normal {
+            data = _ddata_reallocate(data, eltType, oldSize, newSize);
+          } else {
+            data = _ddata_reallocate(data, eltType, oldSize, newSize,
+                                     initElts=false);
```

Comment:
It seems to me that using the enum of Normal / NoInit / Zero as a ddata_reallocate argument would make this simpler

Reply:
So the array would store that enum instead of this `chpl_ArrayResizePolicy` when deciding what to do?

Reply:
I was just talking about changing `_ddata_reallocate` to take an `enum` argument instead of `initElts:bool`. Wasn't asking you to change what the array stores.

Reply:
I think I would rather just have `_ddata_reallocate` take a param to turn initialization on/off. My reasoning is that we use `_ddata` buffers for other things, and I don't really want to expose array specific logic in the signature.

Reply:
Other than the name of the enum, I don't think choosing between "don't init elements", "zero elements", and "default init elements" is in any way array specific.

Reply:
Do we want to make `_ddata_allocate` take this enum as well?

Reply:
That would make sense to me.